recipes/approvaltests.cpp/all/conanfile.py

```diff
@@ -12,22 +13,31 @@ class ApprovalTestsCppConan(ConanFile):
                   "test assertions for each element."
     topics = ("conan", "testing", "unit-testing", "header-only")
     options = {
+        "with_boosttest": [True, False], # Should this be: with_boost_unit_test_framework?
         "with_catch2": [True, False],
         "with_gtest": [True, False],
         "with_doctest": [True, False]
     }
     default_options = {
+        "with_boosttest": False,
         "with_catch2": False,
         "with_gtest": False,
         "with_doctest": False
     }
     no_copy_source = True
 
+    def configure(self):
+        if not self._boost_test_supported():
+            del self.options.with_boosttest
+            self.default_options.__delitem__("with_boosttest")
+
     @property
     def _header_file(self):
         return "ApprovalTests.hpp"
 
     def requirements(self):
+        if self._boost_test_supported() and self.options.with_boosttest:
```

Comment:
```suggestion
        if self.options.get_safe("with_boosttest"):
```

As with_boosttest can be deleted, the method `get_safe` returns None when it's not available, or True/False when available.